src/pages/index.js

```diff
@@ -52,28 +100,54 @@ cardSection.renderItems();
 const editProfilePopup = new PopupWithForm(
   '#profile-edit-modal',
   (formData) => {
-    userInfo.setUserInfo({
-      name: formData.title,
-      description: formData.description
-    });
+    api.setUserInfo(formData.title, formData.description)
+      .then(updatedUserData => {
+        userInfo.setUserInfo({
+          name: updatedUserData.name,
+          description: updatedUserData.about,
+          avatar: updatedUserData.avatar 
+        });
+        editProfilePopup.close();
+      })
+      .catch(err => console.error(err));
   }
 );
 editProfilePopup.setEventListeners();
 
-const addCardPopup = new PopupWithForm(
-  '#profile-add-card',
-  (formData) => {
-    const newCard = createCard({
-      name: formData.title,
-      link: formData.url
+const addCardPopup = new PopupWithForm('#profile-add-card', (formData) => {
+  api.addNewCard(formData.title, formData.url)
+    .then((newCardData) => {
+      const newCardElement = createCard(newCardData);
+      cardSection.addItem(newCardElement);
+      
+      addCardPopup.close();
+      addCardFormValidator.disableButton();
+    })
+    .catch((err) => {
+      console.error('Error adding card:', err);
     });
-    cardSection.addItem(newCard);
-
-    addCardFormValidator.disableButton();
-  }
-);
+});
 addCardPopup.setEventListeners();
 
+
+const avatarPopup = new PopupWithForm('#avatar-edit-modal', (formData) => {
+  api.setAvatar(formData.avatar)
```

Comment:
* When I add a new card, edit the profile data or change the avatar, there is an error https://snipboard.io/boucEA.jpg 


Please, add `return` here to be able to use `.then`, `.catch`., `.finally` inside the `PopupWithForm` class

```js
  (formData) => {
    return api  // here
    .setAvatar(formData.avatar)
```